### PR TITLE
Process relative path from cache config

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
If `path` in cache configuration is not empty and is not absolute path, then it will be put in `<clickhouse server data directory>/caches/<path_from_cache_config>`.

### CODE REVIEW

**Review by tavplubix — APPROVED** [review]

**Comment by kssenii** [comment]
> Fast test — Cannot clone repository

```
docker: error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": EOF.
See 'docker run --help'.
```

:/
cc @Felixoid

**Comment by Felixoid** [comment]
It's an issue with races on fresh setup runners, that aren't in the github API yet. The details are explained in the [thread](https://clickhouse-inc.slack.com/archives/C02N5V7EU57/p1681326818489609)

tl;dr: the termination lambda needs additional requests to AWS API, but I am afraid it could make it slower, so it won't fit the hard 2s timeout. I need time to work on it.
